envoy/http/codec.h

```diff
@@ -227,6 +194,56 @@ class RequestDecoder : public virtual StreamDecoder {
    * @return StreamInfo::StreamInfo& the stream_info for this stream.
    */
   virtual StreamInfo::StreamInfo& streamInfo() PURE;
+
+  /**
+   * @return Shared pointer to the stream_info for this stream.
+   */
+  virtual std::shared_ptr<StreamInfo::StreamInfo> streamInfoSharedPtr() PURE;
+
+  /**
+   * @return List of shared pointers to access loggers for this stream.
+   */
+  virtual std::list<AccessLog::InstanceSharedPtr> accessLogHandlers() PURE;
+};
+
+/**
+ * Stream encoder used for sending a response (server to client). Virtual inheritance is required
+ * due to a parallel implementation split between the shared base class and the derived class.
+ */
+class ResponseEncoder : public virtual StreamEncoder {
```

Comment:
I'm probably missing something but why move this?

Reply:
It's moved lower in the file so that is can refer to RequestDecoder (in the new setRequestDecoder function).

Reply:
WDYT of just pre-declaring so we don't lose the diffs?  If that doesn't work you can just mark this resolved.

Reply:
done, thanks for the suggestion.